db/upgrade.php

```diff
@@ -401,7 +401,6 @@ function xmldb_local_recompletion_upgrade($oldversion) {
         $table->add_field('complete', XMLDB_TYPE_CHAR, '1', null, XMLDB_NOTNULL, null, 'n');
         $table->add_field('grade', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, '0');
         $table->add_field('userid', XMLDB_TYPE_INTEGER, '10', null, null, null, null);
-        $table->add_field('course', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, null);
```

Comment:
please don't modify previous upgrade blocks to remove a field - this should stay here.

Reply:
No worries, was just trying to avoid doing a add + change for all future changes. I'll revert this change.

Reply:
You could look at fixing the default value in that line to prevent new installs ending up with an issue rather than removing the line, but we'd still need the upgrade step later to deal with existing installs anyway.

Reply:
It is a sound suggestion, I've added that. Who knows, maybe Moodle can be smart about it later on and do a no-op for the defaulting if it's already correct.